docs/GettingStarted.md

```diff
@@ -734,11 +735,14 @@ Datadog.tracer.configure(priority_sampling: true)
 
 Once enabled, the sampler will automatically assign a priority of 0 or 1 to traces, depending on their service and volume.
 
-You can also set this priority manually to either drop a non-interesting trace or to keep an important one. For that, set the `context#sampling_priority` to 0 or 2. It has to be done before any context propagation (fork, RPC calls) to be effective:
+You can also set this priority manually to either drop a non-interesting trace or to keep an important one. For that, set the `context#sampling_priority` to `-1` or `2`. It has to be done before any context propagation (fork, RPC calls) to be effective.
+For example, it is possible to select some traces based on some existing bit of information such as a user or transaction ID.
+But it is generally not possible to select a trace based on its error status,
```

Comment:
I *think* we could support that with our current [processing pipeline](https://github.com/DataDog/dd-trace-rb/pull/214).

It would look like:

```rb
Datadog::Pipeline.before_flush do |trace|
  if trace.any? && trace[0].status == Datadog::Ext::Errors::STATUS
    trace[0].set_metric(Datadog::Ext::DistributedTracing::SAMPLING_PRIORITY_KEY, 2)
  end
end
```
I'm not considering here the performance implications of this, but if we want to support this "keep-all-errors" feature, we could very easily perform this pipeline setup.

Reply:
Exactly. The processing pipeline is meant also for that (and we can provide a default pipeline step that users can add if needed). Also, as far as I know, I don't see any performance implication.

Reply:
I don't think the processor would really help here. Because, the thing is not about filtering spans, or sending them or not to the agent. All of them should be submitted, regardless of priority, by default. The fact the processor is doing things before it's sending things to the agent does not help either.

Indeed, consider the following scenario.

* service `a` creates a trace
* sampler decides this should not be kept
* 10 spans are added to the trace
* a call is made to a remove service `b` with distributed sampling headers propagation
* service `b` calls a service `c` with distributed sampling headers propagation
* service `c` is done, no errors, no nothing, so sub-trace is flushed and sent to agent, agent calculates stats on it, and drops the trace
* service `b` is done no errors, no nothing, so sub-trace is flushed and sent to agent, agent calculates stats on it, and drops the trace
* service `a` continues, and there, it's an error
* user sets `context_priority` to `USER_KEEP`
* trace on service `a` is finally done, it's flushed, it's popped (https://github.com/DataDog/dd-trace-rb/pull/214/files#diff-0aa40585cc9b770a220ef7fc86b6655fR29) then it goes through the processing pipeline, marked as an error
* agent receives the trace, calculates stats on it, keeps it, sends it to the backend

You indeed get a trace in the pipeline, marked as "error" and "should be kept" but the problem is: it's incomplete -> the sub traces in service `b` and `c` will never reach our backend, they will be dropped, according to the instruction we gave to the local agents at that time, before the error occurred.